src/anal_ghidra.cpp

```diff
@@ -0,0 +1,3553 @@
+				        "from SLEIGH, abort.",
+				        group.c_str());
+				return nullptr;
+			}
+
+			if(group == r_reg_type_arr[i])
+			{
+				buf << r_reg_string_arr[i] << '\t';
+				break;
+			}
+		}
+
+		buf << sanal.reg_mapping[p->name] << "\t." << p->size * 8 << "\t" << p->offset << "\t"
+		    << "0\n";
+	}
+
+	if(!sanal.pc_name.empty())
+		buf << "=PC\t" << sanal.reg_mapping[sanal.pc_name] << '\n';
+	if(!sanal.sp_name.empty())
+		buf << "=SP\t" << sanal.reg_mapping[sanal.sp_name] << '\n';
+
+	for(unsigned i = 0; i != sanal.arg_names.size() && i <= 9; ++i)
+		buf << "=A" << i << '\t' << sanal.reg_mapping[sanal.arg_names[i]] << '\n';
+
+	for(unsigned i = 0; i != sanal.ret_names.size() && i <= 3; ++i)
+		buf << "=R" << i << '\t' << sanal.reg_mapping[sanal.ret_names[i]] << '\n';
+
+	ut64 pp = 0;
+	string arch = sanal.sleigh_id.substr(pp, sanal.sleigh_id.find(':', pp) - pp);
+	pp = sanal.sleigh_id.find(':', pp) + 1;
+	bool little = sanal.sleigh_id.substr(pp, sanal.sleigh_id.find(':', pp) - pp) == "LE";
+	pp = sanal.sleigh_id.find(':', pp) + 1;
+	int bits = std::stoi(sanal.sleigh_id.substr(pp, sanal.sleigh_id.find(':', pp) - pp));
+	pp = sanal.sleigh_id.find(':', pp) + 1;
+
+	append_hardcoded_regs(buf, arch, little, bits);
+
+	const std::string &res = buf.str();
+	// fprintf(stderr, "%s\n", res.c_str());
+	return strdup(res.c_str());
+}
+
+#define ERR(x)              \
+	if(esil->verbose)       \
+	{                       \
+		eprintf("%s\n", x); \
+	}
+
+constexpr int ESIL_PARM_FLOAT = 127; // Avoid conflict
+
+static bool esil_pushnum_float(RAnalEsil *esil, long double num)
+{
+	char str[64];
+	snprintf(str, sizeof(str) - 1, "%.*LeF", DECIMAL_DIG, num);
+	return r_anal_esil_push(esil, str);
+}
+
+static int esil_get_parm_type_float(RAnalEsil *esil, const char *str)
+{
+	int len, i;
+
+	if(!str || !(len = strlen(str)))
+		return R_ANAL_ESIL_PARM_INVALID;
+
+	if((str[len - 1] == 'F') && (str[1] == '.' || (str[2] == '.' && str[0] == '-')))
+		return ESIL_PARM_FLOAT;
+	if(!strcmp(str, "nanF") || !strcmp(str, "infF") || !strcmp(str, "-nanF") ||
+	   !strcmp(str, "-infF"))
+		return ESIL_PARM_FLOAT;
+
+	return R_ANAL_ESIL_PARM_INVALID;
+}
+
+static long double esil_get_double(RReg *reg, RRegItem *item)
+{
+	RRegSet *regset;
+	float vf = 0.0f;
+	double vd = 0.0f;
+	int off;
+	long double ret = 0.0f;
+	if(!reg || !item)
+	{
+		return 0LL;
+	}
+	off = BITS2BYTES(item->offset);
+	regset = &reg->regset[item->arena];
+	switch(item->size)
+	{
+		case 32:
+			if(regset->arena->size - off - 1 >= 0)
+			{
+				memcpy(&vf, regset->arena->bytes + off, sizeof(float));
+				ret = vf;
+			}
+			break;
+		case 64:
+			if(regset->arena->size - off - 1 >= 0)
+			{
+				memcpy(&vd, regset->arena->bytes + off, sizeof(double));
+				ret = vd;
+			}
+			break;
+		case 80:
+		case 96:
+		case 128:
+			if(regset->arena->size - off - 1 >= 0)
+			{
+				memcpy(&ret, regset->arena->bytes + off, sizeof(long double));
```

Comment:
I do not think using `long double` is a good idea.

Reply:
Not all compilers supports 64+ bits regs. I suggest to limit this to x64 via ifdef endif